tutorials/intermediate/blitting.py

```diff
@@ -0,0 +1,218 @@
+"""
+=================
+Blitting tutorial
+=================
+
+'Blitting' is a `standard technique
+<https://en.wikipedia.org/wiki/Bit_blit>`__ in raster graphics that,
+in the context of Matplotlib, can be used to (drastically) improve
+performance of interactive figures. For example, the
+:mod:`~.animation` and :mod:`~.widgets` modules use blitting
+internally. Here, we demonstrate how to implement your own blitting, outside
+of these classes.
+
+The source of the performance gains is simply not re-doing work we do
+not have to.  If the limits of an Axes have not changed, then there is
+no need to re-draw all of the ticks and tick-labels (particularly
+because text is one of the more expensive things to render).
+
+The procedure to save our work is roughly:
+
+- draw the figure, but exclude any artists marked as 'animated'
+- save a copy of the RBGA buffer
+
+In the future, to update the 'animated' artists we
+
+- restore our copy of the RGBA buffer
+- redraw only the animated artists
+- show the resulting image on the screen
+
+thus saving us from having to re-draw everything which is _not_
+animated.  One consequence of this procedure is that your animated
+artists are always drawn at a higher z-order than the static artists.
+
+Not all backends support blitting.  You can check if a given canvas does via
+the `.FigureCanvasBase.supports_blit` property.
+
+Minimal example
+---------------
+
+We can use the `.FigureCanvasAgg` methods
+`~.FigureCanvasAgg.copy_from_bbox` and
+`~.FigureCanvasAgg.restore_region` in conjunction with setting
+``animated=True`` on our artist to implement a minimal example that
+uses blitting to accelerate rendering
+
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+x = np.linspace(0, 2 * np.pi, 100)
+
+fig, ax = plt.subplots()
+
+# animated=True tells matplotlib to only draw the artist when we
+# explicitly request it
+(ln,) = ax.plot(x, np.sin(x), animated=True)
+
+# make sure the window is raised, but the script keeps going
+plt.show(block=False)
+
+# stop to admire our empty window axes and ensure it is rendered at
+# least once.
+#
+# We need to fully draw the figure at it's final size on the screen
+# before we continue on so that :
+#  a) we have the correctly sized and drawn background to grab
+#  b) we have a cached renderer so that ``ax.draw_artist`` works
+plt.pause(0.1)
```

Comment:
I think the important part is to make sure someone landing here as a starting point to learning about matplotlib interaction needs to know what (semantically) `plt.pause` is doing (spinning the event loop)
```suggestion
# so we spin the event loop to let the backend process any pending operations
plt.pause(0.1)
```

I think with the new comments, this would 100% prevent me from having had any of the confusions that I initially ran into.